### PR TITLE
Create initial potential guess from Boltzmann Inverse of Target RDF

### PR DESCRIPTION
**Pull request by chrisjonesBSU**
This PR adds functionality to create the initial guess pair potentials by taking the Boltzmann inverse of the target RDFs for each pair-state.

It uses a weighted average of each state's target RDF when creating a pair potential. 

To Do:

- [x] Unit Tests
- [ ] Smooth the initial potential

### CODE REVIEW

**Review by jennyfothergill — COMMENTED** [review]
This is a great option to add! Mainly my suggestions relate to how we implement the API. Right now the `potential` passed to `Pair` can either be `None` or a potential array (is it just the y-values?). We could assume if `None` is passed that the initial should be derived from the Boltzmann Inversion of the RDF. Then you could remove the `derive_init_potential` flag. This would move the logic from `optimize` to `Pair`, and remove the need for some of the warnings and error messages.

**Comment by chrisjonesBSU** [comment]
Yeah, I agree. Just pushed those changes.

File: msibi/pair.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Pair(object):
     ----------
     name : str
         Pair name.
-    potential : func
+    potential : 1D numpy array
```
Comment (jennyfothergill) [inline msibi/pair.py:41]: We could add that this can also be None

**Review by jennyfothergill — APPROVED** [review]
Minor suggestion then LGTM

Maybe we should also test when potential=None

**Comment by chrisjonesBSU** [comment]
Just a quick update on this:

It looks like it's working, although I think we might need a couple more things to make it more robust. For example, one of potentials created via Boltzmann inverse shot up almost vertically near the potential cutoff even though the RDF was flat and near 1 there.  The other potentials looked pretty good/normal.

However, regardless of how good/normal the initial potentials look, I'm getting particle out of box errors in hoomd. I'm sitll not sure what is causing that.

**Comment by chrisjonesBSU** [comment]
I'm going to close this for now given the upcoming changes in #27 .  I don't think this feature is urgent. From what I've seen so far, it essentially doesn't matter what your initial potential guess is, so creating better initial guesses (which this would achieve) doesn't seem needed.
